File: app/src/main/java/in/woloo/www/period_tracker/adapters/ShowDailyLogAdapter.java
package in.woloo.www.period_tracker.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;
import java.util.Map;
import java.util.Set;

import butterknife.BindView;
import butterknife.ButterKnife;
import in.woloo.www.R;
import in.woloo.www.dailylogscreen.models.DailyLogSymptoms;
import in.woloo.www.my_history.model.MoodsModel;
import in.woloo.www.period_tracker.model.DailyLogWithTitle;
import in.woloo.www.utils.ImageUtil;

public class ShowDailyLogAdapter extends RecyclerView.Adapter<ShowDailyLogAdapter.ViewHolder> {

    public Context context;
    private List<DailyLogWithTitle> moodsModelsList;

    public ShowDailyLogAdapter(Context context) {
        this.context = context;
    }

    public void addMoods(List<DailyLogWithTitle> moodsModelsList) {
        this.moodsModelsList = moodsModelsList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem = layoutInflater.inflate(R.layout.show_daily_log_data_layout, parent, false);//woloo_search_item
        ShowDailyLogAdapter.ViewHolder viewHolder = new ShowDailyLogAdapter.ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        DailyLogWithTitle dailyLogWithTitle = moodsModelsList.get(position);
        holder.tvMoods.setText(dailyLogWithTitle.getTitleName());
        int symptomsSize = dailyLogWithTitle.getDailyLogSymptoms().size();
        if (symptomsSize == 1) {
            holder.imvMultipleMoods.setVisibility(View.GONE);
        } else {
            holder.imvMultipleMoods.setVisibility(View.VISIBLE);
            holder.imvMultipleMoods.setText("+" + (symptomsSize - 1));
        }
        holder.imvMoods.setImageResource(dailyLogWithTitle.getDailyLogSymptoms().get(0).imageURL);
    }

    @Override
    public int getItemCount() {
        return moodsModelsList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.moods_Tv)
        TextView tvMoods;

        @BindView(R.id.moods_Imv)
        ImageView imvMoods;

        @BindView(R.id.multipleMoods_Imv)
        TextView imvMultipleMoods;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }
}

File: app/src/main/java/in/woloo/www/more/period_tracker/adapters/ShowDailyLogAdapter.kt
package `in`.woloo.www.more.period_tracker.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import `in`.woloo.www.R
import `in`.woloo.www.more.period_tracker.model.DailyLogWithTitle

class ShowDailyLogAdapter(var context: Context) :
    RecyclerView.Adapter<ShowDailyLogAdapter.ViewHolder>() {
    private var moodsModelsList: List<DailyLogWithTitle>? = null

    fun addMoods(moodsModelsList: List<DailyLogWithTitle>?) {
        this.moodsModelsList = moodsModelsList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem = layoutInflater.inflate(
            R.layout.show_daily_log_data_layout,
            parent,
            false
        ) //woloo_search_item
        val viewHolder = ViewHolder(listItem)
        return viewHolder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dailyLogWithTitle: DailyLogWithTitle = moodsModelsList!![position]
        holder.tvMoods!!.setText(dailyLogWithTitle.titleName)
        val symptomsSize: Int = dailyLogWithTitle.getDailyLogSymptoms().size
        if (symptomsSize == 1) {
            holder.imvMultipleMoods!!.visibility = View.GONE
        } else {
            holder.imvMultipleMoods!!.visibility = View.VISIBLE
            holder.imvMultipleMoods!!.text = "+" + (symptomsSize - 1)
        }
        holder.imvMoods!!.setImageResource(dailyLogWithTitle.getDailyLogSymptoms()[0]!!.imageURL)
    }

    override fun getItemCount(): Int {
        return moodsModelsList!!.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @JvmField
        @BindView(R.id.moods_Tv)
        var tvMoods: TextView? = null

        @JvmField
        @BindView(R.id.moods_Imv)
        var imvMoods: ImageView? = null

        @JvmField
        @BindView(R.id.multipleMoods_Imv)
        var imvMultipleMoods: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}
